spec/mongoid/changeable_spec.rb

```diff
@@ -1012,12 +1012,29 @@
         person.user_accounts << user_account
       end
 
-      it 'returns a hash of changes' do
-        pending 'https://jira.mongodb.org/browse/MONGOID-4843'
+      it 'should not add to the changes or changed_attributes hash' do
+        person.changes.should == {}
+        person.changed_attributes.should == {}
+      end
```

Comment:
Foreign key changes on `_ids` must continue to be tracked for both BelongsTo and HasAndBelongsToMany. The correct spec here is:

Reply:
Hi @johnnyshields For referenced associations the `changes` hash was always the empty hash on change to the association, for every mongoid version that I looked at. The spec you provided would actually be a change to the current Mongoid functionality. Here's a code snippet if you'd like to verify that:
```
class User
  include Mongoid::Document

  has_and_belongs_to_many :addresses
end

class Address
  include Mongoid::Document

  has_and_belongs_to_many :users
end

a = Address.create!
u = User.create!

u.addresses << a
p u.changed
```

Reply:
@Neilshweky at least if I use `u.address_ids` for assignment, I see that it does appear in `u.changes`:

```ruby
u.address_ids << a._id
=> [BSON::ObjectId('6286663e071daa69cc155284')]

u.changes
=> {"address_ids"=>[nil, [BSON::ObjectId('6286663e071daa69cc155284')]]}
```

Reply:
ah... ok. I keep forgetting `u.addresses` persists immediately, while `u.address_ids` does not. Hence `u.addresses` doesn't track the change (not even in `previous_changes`).

As long as  `u.address_ids << a._id` continues to track the change, we're fine.

We've discussed in other threads but the immediate persistence of relations when using `<<` is one thing we should really change. It causes so much confusion...

Reply:
Thank you for taking time to make this script!

Reply:
Ah yes, I was talking about the association itself while you were talking about the `_ids` field. Apologies for the confusion... The address_ids does still continue to be populated in the changes hash after this PR.

Reply:
done!